### PR TITLE
Update documentation on I/O

### LINKED ISSUE #212: Update the documentation for reading / writing collections using IOSvc
The [documentation](https://github.com/key4hep/k4FWCore/blob/main/doc/PodioInputOutput.md) currently does not mention the `IOSvc` at all. Since we want people to move to that and, since this is the thing that people see on https://key4hep.github.io/key4hep-doc/how-tos/k4fwcore/doc/PodioInputOutput.html we should probably update that to also include that.

### PR DESCRIPTION
**Pull request by m-fila**
BEGINRELEASENOTES
- Added documentation for reading and writing EDM4hep files with the `IOSvc`
- Moved documentation on `k4DataSvc` to legacy page

ENDRELEASENOTES

Some extensions (namely `sphinx-design` and from myst_parser `colon_fence`) are used to render the pages more nicely in sphinx-based documentation with admonitions and tabs (setting properties either in steering file or from command line):
image

Closes #212

### CODE REVIEW

File: doc/PodioInputOutput.md [hunk 1]
```diff
@@ -18,115 +18,218 @@ limitations under the License.
...
-configure this service like the following
+`IOSvc` is an external Gaudi service for reading and writing EDM4hep and other PODIO-based data models. The service should be imported from `k4FWCore` and named "IOSvc" as other components may look for it under this name.
 
 ```python
-from Gaudi.Configuration import *
-from Configurables import k4DataSvc
+from k4FWCore import IOSvc
 
-evtSvc = k4DataSvc("EventDataSvc")
+io_svc = IOSvc("IOSvc")
```
Comment (tmadlener) [inline doc/PodioInputOutput.md:30]: I don't know if we need / want to mention this explicitly in the documentation but I think `"IOSvc"` is pretty much required as name for the `IOSvc`, right? At least some of the CLI examples below will only work with that. Can we create an empty `IOSvc()` and rely on a default name?

File: README.md [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
-Algorithm to read data from one or multiple input file(s) on disk.
+### Collection Merger
 
-#### PodioOutput
+Algorithm merging multiple collections of the same type into a single collection.
 
-Algorithm to write data to an output file on disk.
+### EventHeaderCreator
+
+Algorithm creating new `edm4hep::EventHeaderCollection` data object.
```
Comment (jmcarcell) [inline README.md:29]:
```suggestion
Algorithm creating a new `edm4hep::EventHeaderCollection` data object.
```

File: doc/LegacyPodioInputOutput.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+# Legacy reading and writing EDM4hep files in Gaudi with the legacy k4DataSvc
+
+:::{caution}
+`k4DataSvc` is a legacy service previously used in K4FWCore for reading and writing data in EDM4hep or other data models based on PODIO.
+
+The currently used service is `IOSvc`, which offers improved streamlined functionality and better support for modern workflows. For detailed documentation on `IOSvc`, refer to [this documentation](IO.md).
```
Comment (jmcarcell) [inline doc/LegacyPodioInputOutput.md:24]: But there isn't an IO.md?

File: doc/PodioInputOutput.md [hunk 4]
```diff
@@ -18,115 +18,218 @@ limitations under the License.
...
+    "keep SimTrackerHits"
+```
+:::
+::::
+
+## Accessing metadata
+
+The k4FWCore provides the `MetadataSvc` that allows accessing user metadata in PODIO-based data-models. There is no need to instantiate the `MetadataSvc` explicitly when using `IOSvc` as `IOSvc` can instantiate it on its own if needed.
+
+When both `Input` and `Output` properties of `IOSvc` are defined, all the metadata originally present in the input will be propagated to the output, possibly adding also any user metadata created during processing.
```
Comment (jmcarcell) [inline doc/PodioInputOutput.md:182]:
```suggestion
When both the `Input` and `Output` properties of `IOSvc` are defined, all the metadata originally present in the input will be propagated to the output, possibly adding also any user metadata created during processing.
```

File: doc/LegacyPodioInputOutput.md (hunk 3 above)
Comment (m-fila) [reply doc/LegacyPodioInputOutput.md]: I forgot to update when renaming the files. should be the `PodioInputOutput.md`

File: doc/PodioInputOutput.md [hunk 5]
```diff
@@ -18,115 +18,218 @@ limitations under the License.
...
+    EvtSel="NONE",
+-    ExtSvc=[podioevent],
++    ExtSvc=[io_svc],
+)
+```
+
+Both functional algorithms and classic algorithms are compatible with either `IOSvc` or `PodioDataSvc`.
+
+The biggest challenge for the transition are the algorithms and services that explicitly request and operate on `PodioDataSvc`. These components are not compatible with `IOSvc` and their internals have to be adapted for the usage with `IOSvc` case by case.
+
```
[the remark this answers is not in the document]
Comment (m-fila) [reply doc/PodioInputOutput.md]: Removed

[the remark this answers is not in the document]
Comment (tmadlener) [reply doc/PodioInputOutput.md]: I think for `k4SimDelphes` we might be able to remove the `PodioDataSvc` (tracking this in https://github.com/key4hep/k4SimDelphes/issues/143). `Gaudi::Algorithm` will work with `IOSvc`, resp. the `EventDataSvc` from Gaudi when simply adding things to the TES, right?

File: README.md [hunk 6]
```diff
@@ -1,25 +1,40 @@
...
-#### k4DataSvc
+| Current | Legacy | Description |
+|---------|--------|-|
+| IOSvc | k4DataSvc | Service handling the PODIO types and collections |
+| Reader | PodioInput | Algorithm to read data from input files on disk. |
+| Writer | PodioOutput | Algorithm to write data to an output file on disk. |
+| MetadataSvc | MetaDataHandle | Service/Handle handling user defined metadata |
 
-Component wrapping the PodioDataService to handle PODIO types and collections.
+See the [documentation](doc/IO.md) for more information.
```
Comment (tmadlener) [inline README.md:19]:
```suggestion
See the [documentation](doc/PodioInputOutput.md) for more information.
```

File: doc/PodioInputOutput.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jmcarcell) [reply doc/PodioInputOutput.md]: Yes, for `Gaudi::Algorithm` nothing should change, but if there is input and output later then it's not that simple.

**Comment by jmcarcell** [comment]
I don't have any more comments. Anything else @m-fila?

**Comment by m-fila** [comment]
I noticed that `outputCommands` is not capitalized as the rest of the `IOSvc` properties.
I'm not sure if you want to give it similar treatment as with `Input` and `Output` properties

Otherwise that's all. Thank you very much for the comments :smile:

**Comment by jmcarcell** [comment]
I had the same thought when I was reading the diffs. I'm not sure, in this case this was literally the same command as it is used with `PodioDataSvc`. At least with `Output` it was a different parameter name. So I don't have a strong opinion on changing it or keeping it as it is.

File: doc/PodioInputOutput.md (hunk 1 above)
Comment (jmcarcell) [reply doc/PodioInputOutput.md]: I checked to make sure and I didn't see any errors when using a different name or no name at all for `IOSvc`. Only the `Reader` and `Writer` retrieve it through a `ServiceHandle`.

**Pull request merged**